frontend/app/routes/_public+/apply+/$id+/tax-filing.tsx

```diff
@@ -0,0 +1,121 @@
+import { useEffect } from 'react';
+
+import { ActionFunctionArgs, LoaderFunctionArgs, json, redirect } from '@remix-run/node';
+import { Form, useActionData, useLoaderData } from '@remix-run/react';
+
+import { faChevronLeft, faChevronRight } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { useTranslation } from 'react-i18next';
+import { z } from 'zod';
+
+import { Button, ButtonLink } from '~/components/buttons';
+import { ErrorSummary, createErrorSummaryItems, hasErrors, scrollAndFocusToErrorSummary } from '~/components/error-summary';
+import { InputRadios } from '~/components/input-radios';
+import { getApplyFlow } from '~/routes-flow/apply-flow';
+import { getTypedI18nNamespaces } from '~/utils/locale-utils';
+import { RouteHandleData } from '~/utils/route-utils';
+
+const i18nNamespaces = getTypedI18nNamespaces('eligibility');
+
+export const handle = {
+  breadcrumbs: [
+    // prettier-ignore
+    { labelI18nKey: 'eligibility:breadcrumbs.canada-ca', to: '/personal-information' },
+    { labelI18nKey: 'eligibility:breadcrumbs.benefits' },
+    { labelI18nKey: 'eligibility:breadcrumbs.dental-coverage' },
+    { labelI18nKey: 'eligibility:breadcrumbs.canadian-dental-care-plan' },
+  ],
+  i18nNamespaces,
+  pageIdentifier: 'CDCP-00XX',
+  pageTitleI18nKey: 'eligibility:tax-filing.page-title',
+} as const satisfies RouteHandleData;
+
+export async function loader({ request, params }: LoaderFunctionArgs) {
+  const applyFlow = getApplyFlow();
+  const { id, state } = await applyFlow.loadState({ request, params });
+
+  return json({ id, state: state.taxFiling2023 });
+}
+
+export async function action({ request, params }: ActionFunctionArgs) {
+  const applyFlow = getApplyFlow();
+  const { id } = await applyFlow.loadState({ request, params });
+
+  const formData = Object.fromEntries(await request.formData());
+  const parsedDataResult = applyFlow.taxFilingSchema.safeParse(formData);
+
+  if (!parsedDataResult.success) {
+    return json({
+      errors: parsedDataResult.error.format(),
+      formData: formData as Partial<z.infer<typeof applyFlow.taxFilingSchema>>,
+    });
+  }
+
+  const sessionResponseInit = await applyFlow.saveState({
+    request,
+    params,
+    state: { taxFiling2023: parsedDataResult.data },
+  });
+
+  if (['FALSE'].includes(parsedDataResult.data.taxFiling2023)) {
```

Comment:
I'm not sure why this is coded as an array.includes(). Am I missing something?

Why not something like this?

``` typescript
  // prettier-ignore
  return parsedDataResult.data.taxFiling2023 === 'FALSE'
      ? redirect(`/apply/${id}/file-your-taxes`, sessionResponseInit)
      : redirect(`/apply/${id}/date-of-birth`, sessionResponseInit);
```

Reply:
Changed return to suggested code (originally it was a multi option from applicant-information which does not fit this case)